development/src/components/peerId_comp.js

```diff
@@ -0,0 +1,74 @@
+'use client'
+import React, { useState } from 'react';
+import Image from 'next/image';
+import { generatePeerIdCharacter } from '../../src/composables/peerIdGenerator';
+
+import closeIconBlack from '../../public/assets/onboardingIcons/close_black.png';
+import closeIconWhite from '../../public/assets/onboardingIcons/close_light.png';
+
+function PeerId() {
+    const [isOpen, setIsOpen] = useState(true);
+    const [peerSessionId, setPeerSessionId] = useState(generatePeerIdCharacter()); 
```

Comment:
setState is asynchronous, this is error prone

try utilizing a useEffect that sets the state to the generatedPeerID on render